test/fluree/db/query/time_travel_test.clj

```diff
@@ -17,37 +17,74 @@
-                                          ;; when queried by ISO-8601 string ONLY
-                                          ;; (o/w it shows up just fine)
-                                          ;:did (fluree.db.did/private->did-map test-utils/default-private-key)}})
-          start        (System/currentTimeMillis)
-          ledger       (test-utils/load-movies conn 500)
-          one-loaded   (util/epoch-ms->iso-8601-str (+ start 400))
-          three-loaded (util/epoch-ms->iso-8601-str (+ start 900))
-          after        (util/epoch-ms->iso-8601-str (+ 60000 (System/currentTimeMillis)))
-          db           (fluree/db ledger)
-          base-query   {:select '{?s [:*]}
-                        :where  '[[?s :rdf/type :schema/Movie]]}
-          one-movie    @(fluree/query db (assoc base-query :t one-loaded))
-          three-movies @(fluree/query db (assoc base-query :t three-loaded))
-          all-movies   @(fluree/query db (assoc base-query :t after))
-          too-early    @(fluree/query db (assoc base-query
-                                           :t (util/epoch-ms->iso-8601-str
-                                                (- start (* 24 60 60 1000)))))]
+          conn                   @(fluree/connect {:method :memory
+                                                   :defaults
+                                                   {:context
+                                                    test-utils/default-context}})
+          ;; if the :did default below is present on the conn
+          ;; (as it is w/ test-utils/create-conn)
+          ;; then the tests below fail at the last check
+          ;; b/c they can't see the last movie transacted
+          ;; when queried by ISO-8601 string ONLY
+          ;; (o/w it shows up just fine)
+          ;:did (fluree.db.did/private->did-map test-utils/default-private-key)}})
+          start-iso              "2022-10-05T00:00:00Z"
+          start                  (util/str->epoch-ms start-iso)
+          three-loaded-millis    (+ start 60000)
+          all-loaded-millis      (+ three-loaded-millis 60000)
+          three-loaded-iso       (util/epoch-ms->iso-8601-str three-loaded-millis)
+          all-loaded-iso         (util/epoch-ms->iso-8601-str all-loaded-millis)
+          after-one-loaded-iso   (util/epoch-ms->iso-8601-str (+ start 5000))
+          after-three-loaded-iso (util/epoch-ms->iso-8601-str (+ three-loaded-millis
+                                                                 5000))
+          after-all-loaded-iso   (util/epoch-ms->iso-8601-str (+ all-loaded-millis
+                                                                 5000))
+          too-early-iso          (util/epoch-ms->iso-8601-str (- start
+                                                                 (* 24 60 60 1000)))
+          ledger                 @(fluree/create conn "iso8601/test")
+          _                      (with-redefs-fn {#'util/current-time-millis
```

Comment:
Definitely not for this pr, but I think the need for `with-redefs*` in tests points to impure functions and those usually mean more headaches with other stuff later. I think the same is the case here because we're computing the time automatically in `stage`. 

Does it make sense to change the api to also expect the current time to be passed in instead of generated automatically?

Reply:
I think it's better if the `commit!` code just does this for you under the hood (vs. you having to supply the correct and correctly-formatted current time to it). If there were a use case for it beyond testing, it could make sense to allow passing in an optional override time to `commit!`, but I don't know if there is such a use case.

Current time is just one of those state-of-the-world sorts of things that seem like a rare good use case for test stubbing. But that's just like, my opinion, man.